### PR TITLE
Alerting: Edit thresholds by handle in timeseries panel

### LINKED ISSUE #38638: Alerting: Use a threshold handle to updated classic condition
In current alerting we show a handle by the graph that the user can use to update the threshold value in the condition, we think users will miss this in unified alerting. 

The plan for making this for unified alerting is that it will only work with the classic condition and it will only update the first of the classic conditions. First iteration will only `Is Above` and `Is Below`, with range thresholds coming later.

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Based on https://github.com/grafana/grafana/pull/38711. Enables setting thresholds by dragging the handle in timeseries panel.

**Which issue(s) this PR fixes**:

Fixes #38638

**Special notes for your reviewer**:

Threshold drag handles are disabled for is outside range and is within range currently. We want to keep this PR contained and only reach feature parity with current alerting.

### CODE REVIEW

**Comment by dprokop** [comment]
@peterholmberg [I've fixed](https://github.com/grafana/grafana/pull/38881/files#diff-deb15ff23587ed0e703c2475cd22ed916657a4da553096a2d70a5e7bfd52c527R39) the react warning mentioned in https://github.com/grafana/grafana/pull/38711#issuecomment-914318219. The underlying issue is in the react-draggable and is "fixed" according to the discussion here: https://github.com/react-grid-layout/react-draggable/issues/390

**Comment by peterholmberg** [comment]
@dprokop aweseome! Thank you!

File: public/app/features/panel/PanelRenderer.tsx [hunk 1]
```diff
@@ -94,11 +99,13 @@ function useFieldOverrides(
   const series = data?.series;
   const fieldConfigRegistry = plugin?.fieldConfigRegistry;
   const theme = useTheme2();
+  const structureRev = useRef(0);
 
   return useMemo(() => {
     if (!fieldConfigRegistry || !fieldConfig || !data) {
       return;
     }
+    structureRev.current = structureRev.current + 1;
```
Comment (dprokop) [inline public/app/features/panel/PanelRenderer.tsx:108]: @mckn @peterholmberg - I've added the structure revision bump over here, whenever the field config changes. It has couple of implications tho. Basically - we do not guarantee referential equality of field configs that are objects( i.e. `hideFrom`) as well as we do not bump the structureRev only in the structure has actually changed. All of the above happens in the PQR that does plenty of stuff to compare the previous vs current data in order to bump or not the structureRev. 

Do you think this simplification here is OK for alerting needs? I would love to talk in the future about how we can re-use PQRs structureRev bump in the PanelRenderer, or alternatively up in the AlertingQueryRunner. This is not a pressing discussion at the moment, but I don't want to unnecessarily compicate the PanelRenderer now with duplicated code of the structureRev calc.

Comment (peterholmberg) [reply public/app/features/panel/PanelRenderer.tsx:108]: Yes it's fine for now. We should revisit implementations of PQRs in the future, maybe post 8.2.

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 2]
```diff
@@ -84,6 +85,9 @@ export class PanelChrome extends Component<Props, State> {
         onAnnotationUpdate: this.onAnnotationUpdate,
         onAnnotationDelete: this.onAnnotationDelete,
         canAddAnnotations: () => Boolean(props.dashboard.meta.canEdit || props.dashboard.meta.canMakeEditable),
+        // TODO: remove, added only for testing now
```
Comment (mckn) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx:88]: Question: something we need to handle in this PR?

**Review by mckn — COMMENTED** [review]
LGTM!

**Review by domasx2 — COMMENTED** [review]
Hey! I gave it a try. If you add just one condition, it works fine, but with two or more things get weird. Multiple handles appear, if you grab one, you can't "release" it 

https://user-images.githubusercontent.com/847684/133244741-5d878e78-a10f-4458-aacc-1b6cebc0e007.mp4

**Comment by peterholmberg** [comment]
Strange @domasx2, I've seen something similar but thought we squashed that!

**Comment by domasx2** [comment]
> Strange @domasx2, I've seen something similar but thought we squashed that!

I had to be more devious, but still managed to break it, sorry :) breaks if you create two classic condition expressions. Seems like a minor edge case though, not sure why you'd actually do it. 

https://user-images.githubusercontent.com/847684/133461431-f58a72b4-2a1a-4151-beb2-8131a07515d2.mp4

**Review by domasx2 — APPROVED** [review]
Found a minor issue where handle breaks if you create a really contrived rule, but works well in general

**Pull request merged**